packages/@aws-cdk/aws-redshift/lib/private/database-query-provider/create-table.ts

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable-next-line import/no-unresolved */
+import * as AWSLambda from 'aws-lambda';
+import { Column } from '../../table';
+import { TableHandlerProps } from '../handler-props';
+import { ClusterProps, executeStatement, makePhysicalId } from './util';
+
+export async function handler(props: TableHandlerProps & ClusterProps, event: AWSLambda.CloudFormationCustomResourceEvent) {
+  const tableName = props.tableName;
+  const tableColumns = props.tableColumns;
+  const clusterProps = props;
+
+  if (event.RequestType === 'Create') {
+    await createTable(tableName, tableColumns, clusterProps);
+    return { PhysicalResourceId: makeNewPhysicalId(tableName, clusterProps, event.RequestId), Data: { tableName: tableName } };
+  } else if (event.RequestType === 'Delete') {
+    await dropTable(tableName, clusterProps);
+    return;
+  } else if (event.RequestType === 'Update') {
+    const { replace } = await updateTable(tableName, tableColumns, clusterProps, event.OldResourceProperties as TableHandlerProps & ClusterProps);
+    const physicalId = replace ? makeNewPhysicalId(tableName, clusterProps, event.RequestId) : event.PhysicalResourceId;
+    return { PhysicalResourceId: physicalId, Data: { tableName: tableName } };
+  } else {
+    /* eslint-disable-next-line dot-notation */
+    throw new Error(`Unrecognized event type: ${event['RequestType']}`);
+  }
+}
+
+function makeNewPhysicalId(tableName: string, clusterProps: ClusterProps, requestId: string): string {
+  return `${makePhysicalId(tableName, clusterProps)}:${requestId}`;
+}
+
+async function createTable(tableName: string, tableColumns: Column[], clusterProps: ClusterProps) {
+  const tableColumnsString = tableColumns.map(column => `${column.name} ${column.dataType}`).join();
+  await executeStatement(`CREATE TABLE ${tableName} (${tableColumnsString})`, clusterProps);
+}
+
+async function dropTable(tableName: string, clusterProps: ClusterProps) {
+  await executeStatement(`DROP TABLE ${tableName}`, clusterProps);
+}
+
+async function updateTable(
+  tableName: string,
+  tableColumns: Column[],
+  clusterProps: ClusterProps,
+  oldResourceProperties: TableHandlerProps & ClusterProps,
+): Promise<{ replace: boolean }> {
+  const oldClusterProps = oldResourceProperties;
+  if (clusterProps.clusterName !== oldClusterProps.clusterName || clusterProps.databaseName !== oldClusterProps.databaseName) {
+    await createTable(tableName, tableColumns, clusterProps);
+    return { replace: true };
+  }
+
+  const oldTableName = oldResourceProperties.tableName;
+  if (tableName !== oldTableName) {
+    await createTable(tableName, tableColumns, clusterProps);
+    return { replace: true };
```

Comment:
hmm. feels somewhat easy to lose data.

Where is the retention policy handled? 
Directly by CloudFormation? (If so, TIL that CFN supported retention policy for custom resources)

Reply:
yep retention is managed in the normal way by CFN! defaults to retain so should be hard to lose data